OpenRA.Mods.Common/Traits/AutoCarryall.cs

```diff
@@ -9,38 +9,70 @@
  */
 #endregion
 
+using System.Collections.Generic;
 using System.Linq;
 using OpenRA.Activities;
 using OpenRA.Mods.Common.Activities;
+using OpenRA.Support;
 using OpenRA.Traits;
 
 namespace OpenRA.Mods.Common.Traits
 {
-	[Desc("Automatically transports harvesters with the Carryable trait between resource fields and refineries.")]
+	[Desc("Automatically transports harvesters with the AutoCarryable and CarryableHarvester between resource fields and refineries.")]
 	public class AutoCarryallInfo : CarryallInfo
 	{
+		[ConsumedConditionReference]
+		[Desc("Boolean expression defining the condition under which the auto carry behavior is enabled. Enabled by default.")]
+		public readonly BooleanExpression AutoCarryCondition = null;
+
 		public override object Create(ActorInitializer init) { return new AutoCarryall(init.Self, this); }
 	}
 
-	public class AutoCarryall : Carryall, INotifyBecomingIdle
+	public class AutoCarryall : Carryall, INotifyBecomingIdle, IObservesVariables
 	{
+		readonly AutoCarryallInfo info;
 		bool busy;
```

Comment:
```suggestion
		bool Busy(Actor self) => self.CurrentActivity != null && self.CurrentActivity is not FlyIdle;
```

By doing this we don't need to hack around to find out wether the carryall is busy

Reply:
If it increase readability, I will fix that.

Reply:
fixed